pkg/params/params.go

```diff
@@ -238,16 +242,60 @@ func (p *Params) ValidateStringMap(cfg map[string]string) error {
 	return nil
 }
 
+func compressAndB64Encode(s string) string {
+	// Create a new zlib.Writer, which will write to a bytes.Buffer
+	var b bytes.Buffer
+	w := zlib.NewWriter(&b)
+
+	// Write the contents of the file to the zlib.Writer
+	if _, err := io.Copy(w, strings.NewReader(s)); err != nil {
+		return "" // fmt.Errorf("failed to copy file %s to zlib.Writer: %w", path, err)
```

Comment:
nit: remove these comments

I don't think these errors are likely to occur (other than on hardware failures or oom situations), but in that case I'd prefer a panic to at least tell you that there's something very bad happening. But that might just be my preference.

Reply:
Agree, in case of hardware failure or oom there are other huge problems/indicators

Reply:
I agree to add a panic. It's better not to hide this error.